arangod/Agency/AgencyComm.cpp

```diff
@@ -1821,19 +1820,43 @@ bool AgencyComm::tryInitializeStructure() {
 }
 
 bool AgencyComm::shouldInitializeStructure() {
-  VPackBuilder builder;
-  builder.add(VPackValue(false));
 
-  // "InitDone" key should not previously exist
-  auto result = casValue("InitDone", builder.slice(), false, 60.0,
-                         AgencyCommManager::CONNECTION_OPTIONS._requestTimeout);
+  while (!application_features::ApplicationServer::isStopping()) {
+    
+    auto result = getValues("Plan");
+
+    if (!result.successful()) { // Not 200 - 299
+      
+      // Agency not ready yet
+      LOG_TOPIC(TRACE, Logger::AGENCYCOMM)
+        << "waiting for agency to become ready";
+      continue;
+      
+    } else {
+
+      // Sanity
+      if (result.slice().isArray() && result.slice().length() == 1) {
+
+        // No plan entry? Should initialise
+        if (result.slice()[0] == VPackSlice::emptyObjectSlice()) {
+          LOG_TOPIC(DEBUG, Logger::AGENCYCOMM)
+            << "agency initialisation should be performed";
+          return true;
+        } else {
+          LOG_TOPIC(DEBUG, Logger::AGENCYCOMM)
+            << "agency initialisation under way or done";
+        }
+      } else {
+        // Should never get here
```

Comment:
Should this not be a FATAL error with a proper log message?

Reply:
Changed somewhat